solutions/src/main/java/ru/sberbank/school/task11/Streams.java

```diff
@@ -0,0 +1,158 @@
+        this.elements.addAll(elements);
+    }
+
+    /**
+     * Принимает на вход массив элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull T... elements) {
+        return new Streams<>(new ArrayList<>(Arrays.asList(elements)));
+    }
+
+    /**
+     * Принимает на вход коллекцию элементов и возвращает стрим построенный на основе этих элементов.
+     *
+     * @param elements входные элементы
+     * @return стрим элементов из elements
+     */
+    public static <T> Streams<T> of(@NonNull Collection<T> elements) {
+        return new Streams<>(new ArrayList<>(elements));
+    }
+
+    /**
+     * Фильтрация элементов по заданному правилу. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Иными словами, этот метод оставляет в коллекции только те элементы, которые удовлетворяют условию в лямбде.
+     * <p>
+     * Аналог Stream#map(Function).
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило фильтрации элементов.
+     * @return стрим
+     */
+    public Streams<T> filter(Predicate<? super T> object) {
+        suppliers.add(() -> elements.stream()
+                .filter(object)
+                .collect(Collectors.toList()));
+        return this;
+    }
+
+    /**
+     * Преобразование элементов в какие-то другие элементы. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Например, данный метод должен уметь извлекать какие-то поля из объектов исходного типа (если это возможно).
+     * <p>
+     * Аналог Stream#map.
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило траснформации элементов.
+     * @return стрим
+     */
+    @SuppressWarnings("unchecked")
+    public <R> Streams<R> transform(Function<? super T, ? extends R> object) {
+
+        List<R> temp = new ArrayList<>();
+
+        suppliers.add(() -> {
+            temp.addAll(elements.stream()
+                    .map(object)
+                    .collect(Collectors.toList()));
+            return (List<T>) temp;
+        });
+
+        return (Streams<R>) this;
+    }
+
+    /**
+     * Сортировка элементов. Необходимо подобрать нужый интерфейс для передачи в этот метод.
+     * <p>
+     * Данный метод должен уметь сортировать элементы по заданному правилу.
+     * <p>
+     * Аналог Stream#sorted
+     * <p>
+     * Intermediate операция.
+     *
+     * @param object - правило сортировки элементов.
+     * @return стрим
+     */
+    public Streams<T> sorted(Comparator<? super T>  object) {
+        suppliers.add(() -> elements.stream()
+                .sorted(object)
+                .collect(Collectors.toList()));
+        return this;
+    }
+
+    /**
+     * Преобразование стрима в Map. Необходимо подобрать нужные интерфейсы для передачи в этот метод.
+     * <p>
```

Comment:
&lt;p&gt; tag should be preceded with an empty line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a2094476062>*